lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -2225,6 +2225,245 @@ std::optional<size_t> InstanceOp::getTargetResultIndex() {
   return std::nullopt;
 }
 
+// -----------------------------------------------------------------------------
+// InstanceChoiceOp
+// -----------------------------------------------------------------------------
+
+void InstanceChoiceOp::build(
+    OpBuilder &builder, OperationState &result, FModuleLike defaultModule,
+    ArrayRef<std::pair<OptionCaseOp, FModuleLike>> cases, StringRef name,
+    NameKindEnum nameKind, ArrayRef<Attribute> annotations,
+    ArrayRef<Attribute> portAnnotations, StringAttr innerSym) {
+  // Gather the result types.
+  SmallVector<Type> resultTypes;
+  resultTypes.reserve(defaultModule.getNumPorts());
+  llvm::transform(
+      defaultModule.getPortTypes(), std::back_inserter(resultTypes),
+      [](Attribute typeAttr) { return cast<TypeAttr>(typeAttr).getValue(); });
```

Comment:
Nit: this could probably avoid the transform with map in the initializer list of `resultTypes`.